### ISSUE TITLE
Please help me brother (MongoDB issue) error:- cannot decode string into an integer type

### ISSUE THREAD

**Issue by golmujik** [iegomez/mosquitto-go-auth#26]
I followed everything from the readme and I almost done set up everything including mosquitto build, Go-Auth plugin build and all.

But when I linked the Mongo DB and try to authenticate I get an error which I got from the log of mosquitto
`time="2019-07-18T13:24:49+05:30" level=info msg="Cleaning up plugin"
time="2019-07-18T13:24:50+05:30" level=info msg="Backend registered: Mongo"
time="2019-07-18T13:24:50+05:30" level=info msg="No cache set."
time="2019-07-18T13:24:52+05:30" level=debug msg="checking user (*username) with backend Mongo"
time="2019-07-18T13:24:52+05:30" level=debug msg="Mongo get user error: cannot decode string into an integer type"`

below are my settings in config files.

### default.conf
`allow_anonymous false`

### moquitto.conf
`include_dir /etc/mosquitto/conf.d`

### In go-config.conf
`auth_plugin /etc/mosquitto/go-auth.so
auth_opt_backends mongo

auth_opt_mongo_host localhost
auth_opt_mongo_port 27017
auth_opt_mongo_dbname go-auth
#auth_opt_mongo_username ""
#auth_opt_mongo_password ""
auth_opt_mongo_users users
auth_opt_mongo_acls acls

auth_opt_log_level debug
auth_opt_log_dest file
auth_opt_log_file /var/log/mosquitto/mosquitto.log

auth_opt_check_prefix false`

Waiting. :)

**Comment by golmujik** [comment]
In the cluster the DB name is go-auth and the collections are users, & acls

the document in the users collection is here 

`{"_id":"5a4e760f708ba1a1601fa40f","username":"test","password":"$6$ZdpM/sdypubRHD6k$ZXJsi65/9iPvvIrnF3FDvs8tLOBIQ6T6TFolEvalszyIr7pxv1y2G2k6yBWpbHr50kf3BmeXrKTnGSIIIHw3Wg==","superuser":true,"acls":[{"topic":"test/topic/1","acc":1},{"topic":"single/topic/+","acc":1},{"topic":"hierarchy/#","acc":1},{"topic":"write/test","acc":2},{"topic":"test/readwrite/1","acc":3}]}`

** and important thing maybe is my acls collection is empty as I dont want any common acls.

**Comment by golmujik** [comment]
I have an update:-
When I replaced `123456` with `$6$ZdpM/sdypubRHD6k$ZXJsi65/9iPvvIrnF3FDvs8tLOBIQ6T6TFolEvalszyIr7pxv1y2G2k6yBWpbHr50kf3BmeXrKTnGSIIIHw3Wg==`
in the password object in database, I stopped getting the previous error and get this new error when I subscribe or publish any thing. When I check with `systemctl status mosquitto.service`

Jul 18 14:15:11 user systemd[1]: mosquitto.service: Start request repeated too quickly.
Jul 18 14:15:11 user systemd[1]: mosquitto.service: Failed with result 'core-dump'.

**Comment by iegomez** [comment]
Hi. I’m on vacation and don’t have a computer with me so there’s little I can do, but I suggest you check the logs (e.g., sudo systemctl -u mosquitto -f -n 50) or try running mosquitto manually to see what’s crashing.

**Comment by golmujik** [comment]
Hi there @iegomez. A big thank you for replying 🥇  :) 

Even when you are on vacation, means a lot for me :)

I checked the logs of Mosquitto and Mongo and got that the error was because of the plugin's `main.CheckBackendsAuth()` function which check for the hashed password & as I said before I replaced `$6$ZdpM/sdypubRHD6k$ZXJsi65/9iPvvIrnF3FDvs8tLOBIQ6T6TFolEvalszyIr7pxv1y2G2k6yBWpbHr50kf3BmeXrKTnGSIIIHw3Wg==` this with `123456` without hash. 

Although the password I used before using this simple string of `123456` was of Mosquitto generated by the `mosquitto_passwd ....` command so it (plugin) directly failed to check for this unknown hashed.

Then I went to the Go-Auth plugin folder I was a pw file and used it to generate the correct password and replaced it with the Mosquitto's generated password with the plugin's generated password and it solved everything 💯.
